docs/reference/glossary.md

```diff
@@ -0,0 +1,161 @@
+
+## Storage miner
+
+A _storage miner_ is a Filecoin participant that stores data on behalf of the network. Storage miners are rewarded for this service through payments by clients that contract their services, as well as by periodic authorization to extend the Filecoin [blockchain](#blockchain) with [blocks](#block) of their own creation. When they create a block, storage miners are rewarded with newly minted filecoin, and by the transaction fees they can levy on other participants seeking to include [messages](#message) in the block.
+
+## Storage power
+
+A [storage miner's](#storage-miner) _storage power_ is a value roughly proportional to the amount of storage capacity they make available on behalf of the network via [capacity commitments](#capacity-commitment) or [storage deals](#deal). Storage power is used to select storage miners for rewards in proportion to their contributions to the total network storage capacity.
+
+## Zero-knowledge succinct non-interactive argument of knowledge (zk-SNARK)
+
+An _argument of knowledge_ is a construction by which one party, called the _prover_, can convince another, the _verifier_, that the prover has access to some piece of information. There are several possible constraints on such constructions:
+
+- A _non-interactive_ argument of knowledge has the requirement that just a single message, sent from the prover to the verifier, should serve as a sufficient argument.
+
+- A _zero-knowledge_ argument of knowledge has the requirement that the verifier should not need access to the knowledge the prover has access to in order to verify the prover's claim.
+
+- A _succinct_ argument of knowledge is one that can be "quickly" verified, and which is "small", for appropriate definitions of both of those terms.
+
+A zero-knowledge, succinct non-interactive argument of knowledge (zk-SNARK) embodies all of these properties. Filecoin utilizes these constructions to enable its distributed network to efficiently verify that [storage miners](#storage-miner) are storing files they pledged to store, without requiring the verifiers to maintain copies of these files themselves.
+
+## Testnet
+
+A portmanteau of "test" and "network, _testnet_ is a term used to refer to Filecoin networks that are set up for the purposes of development, experimentation, and assessment.
+
+Note: if used as a proper noun, capitalize the term. For example, "I am mining on Testnet."
+
+## Tipset
+
+Filecoin utilizes a novel [blockchain](#blockchain) variant called a [_tipset_](https://filecoin.io/blog/tipsets-family-based-approach-to-consensus/). Tipsets are less strict chains than directed, acyclic graphs; this construction allows multiple blocks to be mined at the same [height](#block-height).
```

Comment:
Would be nice to mention how these groups/families are processed here, ie some mention of weighting and why this is more efficient.

Reply:
Added a lengthier discussion to address these points!